.github/workflows/build-manual.yml

```diff
@@ -0,0 +1,65 @@
+name: Manual Build and Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      os:
+        description: 'OS'
+        required: false
+        default: 'ubuntu-latest'
+        type: choice
+        options:
+          - ubuntu-latest
+          - windows-latest
+      jdk-distribution:
+        description: 'JDK Distribution'
+        required: false
+        default: 'temurin'
+        type: choice
+        options:
+          - temurin
+          - semeru
+          - microsoft
+          - oracle
+          - zulu
+          - corretto
+          - liberica
+      jdk-version:
+        description: 'JDK Version'
+        required: false
+        default: '11'
+        type: choice
+        options:
+          - 11
+          - 17
+          - 18
+      args:
+        description: 'Arguments like -Dquickly | -DskipTests | -DallTests'
```

Comment:
I realize this is an example, but I'm not sure if this will work with `-DallTests`. If it does, the `timeout-minutes` of 120 is definitely not enough time.

Reply:
It should be able to run with -DallTests, with a proper timeout value. But if we make the timeout very long like 4 hours, it starts to lose meaning for other more common use cases like a quick build. How about make timeout configurable with a default value 120?

Reply:
IIRC there are some tests that don't pass with GitHub Actions, however it's been a long time since I ran it. Making it configurable seems reasonable either way.